hovancik/stretchly
# Feature/1457 add shorcuts for pause breaks

Issue: #1457


### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.



- [x]  issue was opened to discuss proposed changes before starting implementation. It is important do discuss changes before implementing them (Why should we add it? How should it work? How should it look? Where will it be? ...).
- [x]  during development, `node` version specified in `package.json` was used (ie using [nvm](https://github.com/creationix/nvm)).
- [x]  package versions and package-lock.json were not changed (`npm install --no-save`).
- [x]  app version number was not changed.
- [x]  all new code has tests to ensure against regressions.
- [x] `npm run lint` reports no offenses.
- [x] `npm run test` is error-free.
- [x]  README and CHANGELOG were updated accordingly.
- []  after PR is approved, all commits in it are [squashed](https://gitbetter.substack.com/p/how-to-squash-git-commits)

### Description of the Change

* Added ability to set keyboard shortcuts to pause the breaks for the period of time (30 min, 1 hr etc.)
* Moved the code for keyboard chortcut registration from `main.js` into `utils/breakShortcuts.js`.

### Verification Process

* Open About tab in the Preferences screen.
* Click Ctrl-D to see the path to preferences JSON
* Set the keyboard shortcuts:

```
"pauseBreaksFor30MinutesShortcut": "CmdOrCtrl+Alt+0",
"pauseBreaksFor1HourShortcut": "CmdOrCtrl+Alt+1",
"pauseBreaksFor2HoursShortcut": "CmdOrCtrl+Alt+2",
"pauseBreaksFor5HoursShortcut": "CmdOrCtrl+Alt+5",
"pauseBreaksUntilMorningShortcut": "CmdOrCtrl+Alt+M",
```

* Restart the app.
* Use the keyboard shortcuts and see that the breaks are paused: the app icon changes to Pause and when you click the text shows the correct pause duration.

* Since I moved the code for other break shortcuts into, I also tested these shortcuts:
```
"pauseBreaksToggleShortcut": "CmdOrCtrl+Alt+T",
"skipToNextScheduledBreakShortcut": "CmdOrCtrl+Alt+B",
"skipToNextMiniBreakShortcut": "CmdOrCtrl+Alt+S",
"skipToNextLongBreakShortcut": "CmdOrCtrl+Alt+L",
"resetBreaksShortcut": "CmdOrCtrl+Alt+R",
```

* I tested it on Mac, Windows and Ubuntu.
